### ISSUE TITLE
 unicode problem

### ISSUE THREAD

**Issue by TheMasterOfMagic** [sinclairtarget/um#22]
I edited the page like this
![image](https://user-images.githubusercontent.com/25054321/47206825-3289bc80-d3bc-11e8-8229-d821dc2b5a97.png)
and when I read it, it outputs like this
![image](https://user-images.githubusercontent.com/25054321/47206874-56e59900-d3bc-11e8-8b5a-5e50076f43e1.png)
am I missing something or um dose not supported unicode character now?

**Comment by sinclairtarget** [comment]
Yeah, it's entirely possible that Um itself messed this up. It's also possibly that the man program doesn't know what to do with unicode in a .roff file. Will look into it.

Thanks!

**Comment by ernstki** [comment]
> Yeah, it's entirely possible that Um itself messed this up. It's also possibly that the man program doesn't know what to do with unicode in a .roff file. Will look into it.

Not your fault. The default `groff` on macOS doesn't support Unicode ([ref][1]).

With a recent `groff` from MacPorts or Homebrew, you can get proper display of UTF-8 symbols with something like this:

```bash
kramdown -i markdown ~/.um/pages/shell/rsync.md -o man | groff -k -Tutf8 -man | less
```

…but that kind of defeats the purpose of `um`. The `-k` option is key here, but that option is not present in the `groff` that comes with macOS, which is too old.

Not sure the way forward for `um`, because this isn't necessarily a problem with Kramdown or `um`, it's the OS-default `groff` that's the obstacle.

[1]: https://apple.stackexchange.com/questions/341783/whats-the-simplest-way-to-process-files-with-utf-8-characters-in-groff-on-macos
[2]: https://github.com/gettalong/kramdown/issues/38#issuecomment-14584000
